src/dvc_objects/db.py

```diff
@@ -59,6 +60,19 @@ def get(self, oid: str) -> Object:
             oid,
         )
 
+    def write(self, oid: str, data: Union[bytes, BinaryIO]) -> None:
+        if isinstance(data, bytes):
+            fobj: "BinaryIO" = BytesIO(data)
+            size: "Optional[int]" = len(data)
+        else:
+            fobj = data
+            size = cast("Optional[int]", getattr(fobj, "size", None))
+
+        path = self.oid_to_path(oid)
+        parent = self.fs.path.parent(path)
+        self.makedirs(parent)
+        self.fs.put_file(fobj, path, size=size)
```

Comment:
Seems like this is more like `add_fobj/bytes`(our current `add` is like `add_path` or `add_from_path`) or something, as you can't (and you shouldn't) write to the same file again.

Also looks like it is not atomic.